### DISCUSSION TITLE
does the client know it's name in meshcentral?

### DISCUSSION THREAD

**Discussion by ne0YT** [Ylianst/MeshCentral#6482] [General]
is it possible to somehow read the name and group which was given in meshcentral from the agent side in a scripted manner?
I use mesh central, but now also need a tool to deploy scripts / do maintenance in a more automated / distributed way.
I'd like to make it easy for myself to findthe same host on both server..

**Comment by si458** [comment]
You can read/phase the .msh file that gets generated as this includes the meshid and meshname

But also if I remember it's listed in the windows registry 
https://github.com/Ylianst/MeshCentral/issues/1455#issuecomment-639281494

**Comment by ne0YT** [comment]
that's not what I'm looking for.
I need the given name which is shown under my Devices in meshcentral

**Reply by si458** [reply]
Oh, you mean the Device Name!
Where u can customise it to be something else
By default, it's just the computers name, but sadly no way of getting it I'm afraid without connecting to the websocket then logging in/authorising then querying ur device info to see what name u gave it

**Reply by ne0YT** [reply]
no way to ask using the meshagent?

**Reply by si458** [reply]
Not that I'm aware off I'm sorry, I will have to look into the source code and see what we can do
